File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/ShaTheHiddenGod.java
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper;
import java.util.LinkedHashMap;

/* JADX INFO: loaded from: classes3.dex */
public class ShaTheHiddenGod extends Enemy {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMaxDamage() {
        return 7;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMinDamage() {
        return 7;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isMagic() {
        return true;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isRanged() {
        return true;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected void configureStatistics() {
        this.baseMaxHp = 7777;
        this.baseConstitution = 7;
        this.baseIntelligence = 7;
        this.baseDexterity = 7;
        this.baseDefense = 0;
        this.baseMagicDefense = 0;
        this.alwaysHits = true;
        this.immunityToStatus = 1.0d;
        this.imageId = R.drawable.unit_sha_the_hidden_god;
        this.idName = R.string.enemy_sha_the_hidden_god_name;
        this.idDescription = R.string.enemy_sha_the_hidden_god_description;
        this.passiveSkill = Skills.PASSIVE_PERFECT_IMMUNITY;
        this.activeSkill = Skills.ACTIVE_DISEMBODY;
        this.expGiven = 9600;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    public LinkedHashMap<ItemWrapper, Integer> listDrops(int i) {
        LinkedHashMap<ItemWrapper, Integer> linkedHashMap = new LinkedHashMap<>();
        linkedHashMap.put(ItemWrapper.getInstance("DivineZygote", 1), 1000);
        return linkedHashMap;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/ShaTheHiddenGod.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper
import java.util.LinkedHashMap

class ShaTheHiddenGod : Enemy() {
    override fun getMaxDamage(): Int = 7
    override fun getMinDamage(): Int = 7
    override fun isMagic(): Boolean = true
    override fun isRanged(): Boolean = true

    override fun configureStatistics() {
        baseMaxHp = 7777
        baseConstitution = 7
        baseIntelligence = 7
        baseDexterity = 7
        baseDefense = 0
        baseMagicDefense = 0
        alwaysHits = true
        immunityToStatus = 1.0
        imageId = R.drawable.unit_sha_the_hidden_god
        idName = R.string.enemy_sha_the_hidden_god_name
        idDescription = R.string.enemy_sha_the_hidden_god_description
        passiveSkill = Skills.PASSIVE_PERFECT_IMMUNITY
        activeSkill = Skills.ACTIVE_DISEMBODY
        expGiven = 9600
    }

    override fun listDrops(i: Int): LinkedHashMap<ItemWrapper, Int> {
        val linkedHashMap = LinkedHashMap<ItemWrapper, Int>()
        linkedHashMap.put(ItemWrapper.getInstance("DivineZygote", 1), 1000)
        return linkedHashMap
    }
}
